core/rawdb/accessors_chain.go

```diff
@@ -1851,6 +1856,22 @@ func WriteCoinbaseLockupToMap(coinbaseMap map[[CoinbaseLockupKeyLength]byte][]by
 	return nil
 }
 
+func WriteCoinbaseLockupToSlice(amount *big.Int, blockHeight uint32, elements uint16, delegate common.Address) ([]byte, error) {
+	data := make([]byte, 38)
+	amountBytes := amount.Bytes()
+	if len(amountBytes) > 32 {
+		return nil, fmt.Errorf("amount is too large")
+	}
+	// Right-align amountBytes in data[:32]
+	copy(data[32-len(amountBytes):32], amountBytes)
```

Comment:
we might need a sanity check for amount bytes == 0, it will panic, if that somehow happens

Reply:
It will write all zeroes to 32 bytes if amountBytes == 0